common/models/access-token.js

```diff
@@ -79,6 +79,119 @@ module.exports = function(AccessToken) {
+   * @property {Boolean} [bearerTokenBase64Encoded] Defaults to `true`. For `Bearer` token based `Authorization` headers,
+   * decode the value from `Base64`. If set to `false`, the decoding will be skipped and the token id will be the raw value
+   * parsed from the header.
+   * @return {String} The access token
+   */
+  AccessToken.getAccessTokenId = function(req, options) {
+    options = options || {};
+    var params = options.params || [];
+    var headers = options.headers || [];
+    var cookies = options.cookies || [];
+    var i = 0;
+    var length, id;
+
+    // https://github.com/strongloop/loopback/issues/1326
+    if (options.searchDefaultTokenKeys !== false) {
+      params = params.concat(['access_token']);
+      headers = headers.concat(['X-Access-Token', 'authorization']);
+      cookies = cookies.concat(['access_token', 'authorization']);
+    }
+
+    for (length = params.length; i < length; i++) {
+      var param = params[i];
+      // replacement for deprecated req.param()
+      id = req.params && req.params[param] !== undefined ? req.params[param] :
+        req.body && req.body[param] !== undefined ? req.body[param] :
+        req.query && req.query[param] !== undefined ? req.query[param] :
+        undefined;
+
+      if (typeof id === 'string') {
+        return id;
+      }
+    }
+
+    for (i = 0, length = headers.length; i < length; i++) {
+      id = req.header(headers[i]);
+
+      if (typeof id === 'string') {
+        // Add support for oAuth 2.0 bearer token
+        // http://tools.ietf.org/html/rfc6750
+        if (id.indexOf('Bearer ') === 0) {
+          id = id.substring(7);
+          if (options.bearerTokenBase64Encoded) {
+            // Decode from base64
+            var buf = new Buffer(id, 'base64');
+            id = buf.toString('utf8');
+          }
+        } else if (/^Basic /i.test(id)) {
+          id = id.substring(6);
+          id = (new Buffer(id, 'base64')).toString('utf8');
+          // The spec says the string is user:pass, so if we see both parts
+          // we will assume the longer of the two is the token, so we will
+          // extract "a2b2c3" from:
+          //   "a2b2c3"
+          //   "a2b2c3:"   (curl http://a2b2c3@localhost:3000/)
+          //   "token:a2b2c3" (curl http://token:a2b2c3@localhost:3000/)
+          //   ":a2b2c3"
+          var parts = /^([^:]*):(.*)$/.exec(id);
+          if (parts) {
+            id = parts[2].length > parts[1].length ? parts[2] : parts[1];
+          }
+        }
+        return id;
+      }
+    }
+
+    if (req.signedCookies) {
+      for (i = 0, length = cookies.length; i < length; i++) {
+        id = req.signedCookies[cookies[i]];
+
+        if (typeof id === 'string') {
+          return id;
+        }
+      }
+    }
+    return null;
+  };
+
+  /**
+   * Validate the access token
+   * @param {String} id Access token
+   * @callback {Fuction} cb Callback function
+   */
+  AccessToken.validate = function(id, cb) {
```

Comment:
- the signature of this new api is inconsistent with `.prototype.validate()` which returns a **Boolean**, i suggest returning a boolean, as the token is already provided as param.
- please add also the @callback signature in the jsdoc.
```
// as in
  @callback (Function) callback
  @param {Error} err
  @param {isValid} is the token valid
```
- you also have a typo (see: _Fuction_)